source/state_representation/src/Space/Cartesian/CartesianPose.cpp

```diff
@@ -1,238 +1,154 @@
-		double period = dt.count();
-		period /= 1e9;
-		// set linear velocity
-		twist.set_linear_velocity(pose.get_position() / period);
-		// set angular velocity from the log of the quaternion error
-		Eigen::Quaterniond log_q = MathTools::log(pose.get_orientation());
-		if(pose.get_orientation().dot(log_q) < 0) log_q = Eigen::Quaterniond(-log_q.coeffs());
-		twist.set_angular_velocity(2 * log_q.vec() / period);
-		return twist;
-	}
-
-	const std::vector<double> CartesianPose::to_std_vector() const
-	{
-		std::vector<double> pose = std::vector<double>(this->get_position().data(), this->get_position().data() + 3);
-		pose.resize(7);
-		pose[3] = this->get_orientation().w();
-		pose[4] = this->get_orientation().x();
-		pose[5] = this->get_orientation().y();
-		pose[6] = this->get_orientation().z();
-		return pose;
-	}
-
-	void CartesianPose::from_std_vector(const std::vector<double>& value)
-	{
-		if (value.size() == 3)
-		{
-			this->set_position(value);
-		}
-		else if (value.size() == 4)
-		{
-			this->set_orientation(value);
-		}
-		else if (value.size() == 7)
-		{
-			this->set_pose(value);
-		}
-		else
-		{
-			throw IncompatibleSizeException("The input vector is of incorrect size");
-		}
-	}
+namespace StateRepresentation {
+CartesianPose::CartesianPose() {}
+
+CartesianPose::CartesianPose(const std::string& name, const std::string& reference) : CartesianState(name, reference) {}
+
+CartesianPose::CartesianPose(const std::string& name, const Eigen::Vector3d& position, const std::string& reference) : CartesianState(name, reference) {
+  this->set_position(position);
+}
+
+CartesianPose::CartesianPose(const std::string& name, const double& x, const double& y, const double& z, const std::string& reference) : CartesianState(name, reference) {
+  this->set_position(x, y, z);
+}
+
+CartesianPose::CartesianPose(const std::string& name, const Eigen::Vector3d& position, const Eigen::Quaterniond& orientation, const std::string& reference) : CartesianState(name, reference) {
+  this->set_position(position);
+  this->set_orientation(orientation);
+}
+
+CartesianPose::CartesianPose(const CartesianPose& pose) : CartesianState(pose) {}
+
+CartesianPose::CartesianPose(const CartesianState& state) : CartesianState(state) {}
+
+CartesianPose::CartesianPose(const CartesianTwist& twist) : CartesianState(std::chrono::seconds(1) * twist) {}
+
+CartesianPose CartesianPose::Identity(const std::string& name, const std::string& reference) {
+  return CartesianPose(name, Eigen::Vector3d::Zero(), Eigen::Quaterniond::Identity(), reference);
+}
+
+CartesianPose CartesianPose::Random(const std::string& name, const std::string& reference) {
+  return CartesianPose(name, Eigen::Vector3d::Random(), Eigen::Quaterniond::UnitRandom(), reference);
+}
+
+CartesianPose& CartesianPose::operator=(const CartesianState& state) {
+  this->CartesianState::operator=(state);
+  return (*this);
+}
+
+Eigen::Vector3d CartesianPose::operator*(const Eigen::Vector3d& vector) const {
```

Comment:
This might be a source of errors because we allow the user to multiply vectors that might be incompatible in terms of reference frames

Reply:
Yes I agree but as opposed to a vector used to represent a pose I think here this make sense. There was cases where I wanted to transform a single vector and not a full pose